pkg/driver/agent.go

```diff
@@ -147,37 +145,17 @@ func (ns *node) NodeUnpublishVolume(
 	volumeID := req.GetVolumeId()
 
 	if vol, err = zfs.GetZFSVolume(volumeID); err != nil {
-		return nil, err
-	}
-
-	if devpath, err = zfs.GetVolumeDevPath(vol); err != nil {
-		goto NodeUnpublishResponse
-	}
-
-	currentMounts, err = zfs.GetMounts(devpath)
-	if err != nil {
-		return nil, err
-	} else if len(currentMounts) == 0 {
-		return nil, status.Error(codes.Internal, "umount request for not mounted volume")
-	} else if len(currentMounts) == 1 {
-		if currentMounts[0] != targetPath {
```

Comment:
I think this condition check went away with the refactored code. Is this condition not required anymore?

Reply:
We are already checking in UmountVolume that if volume is mounted at the targetpath then only unmount it, otherwise we should always return 0 OK.